File: SVM_Frontend/migrate-colors.py
import os
import re

# Color Mappings (Hex to Tailwind Alias or CSS Variable)
mappings = {
    r'#C8102E': 'var(--color-primary)', # Original Hex
    r'#A60D26': 'var(--color-primary-hover)',
    r'#0A0A0B': 'var(--color-bg-default)',
    r'#121214': 'var(--color-bg-paper)',
    r'#1A1A1C': 'var(--color-bg-alt)',
    r'#FFFFFF': 'var(--color-text-primary)',
    r'#D1D1D1': 'var(--color-text-secondary)',
    r'#888888': 'var(--color-text-dim)',
    
    # Common Tailwind Aliases (for bulk replacement in className strings)
    r'bg-\[#0A0A0B\]': 'bg-background',
    r'bg-\[#121214\]': 'bg-background-paper',
    r'bg-\[#1A1A1C\]': 'bg-background-alt',
    r'text-\[#C8102E\]': 'text-primary',
    r'border-\[#C8102E\]': 'border-primary',
}

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original_content = content
    for pattern, replacement in mappings.items():
        content = re.sub(pattern, replacement, content, flags=re.IGNORECASE)
    
    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated: {filepath}")

def main():
    src_dir = os.path.join(os.getcwd(), 'src')
    for root, dirs, files in os.walk(src_dir):
        for file in files:
            if file.endswith(('.js', '.jsx', '.css')):
                process_file(os.path.join(root, file))

if __name__ == '__main__':
    main()

File: SVM_Frontend/migrate-colors.js
const fs = require('fs');
const path = require('path');

const mappings = [
    { pattern: /#C8102E/gi, replacement: 'var(--color-primary)' },
    { pattern: /#A60D26/gi, replacement: 'var(--color-primary-hover)' },
    { pattern: /#0A0A0B/gi, replacement: 'var(--color-bg-default)' },
    { pattern: /#121214/gi, replacement: 'var(--color-bg-paper)' },
    { pattern: /#1A1A1C/gi, replacement: 'var(--color-bg-alt)' },
    { pattern: /#FFFFFF/gi, replacement: 'var(--color-text-primary)' },
    { pattern: /#D1D1D1/gi, replacement: 'var(--color-text-secondary)' },
    { pattern: /#888888/gi, replacement: 'var(--color-text-dim)' },
    
    // Tailwind Class Patterns
    { pattern: /bg-\[#0A0A0B\]/gi, replacement: 'bg-background' },
    { pattern: /bg-\[#121214\]/gi, replacement: 'bg-background-paper' },
    { pattern: /bg-\[#1A1A1C\]/gi, replacement: 'bg-background-alt' },
    { pattern: /text-\[#C8102E\]/gi, replacement: 'text-primary' },
    { pattern: /border-\[#C8102E\]/gi, replacement: 'border-primary' }
];

function processFile(filePath) {
    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;

    mappings.forEach(m => {
        content = content.replace(m.pattern, m.replacement);
    });

    if (content !== original) {
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Updated: ${filePath}`);
    }
}

function walk(dir) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            walk(fullPath);
        } else if (file.endsWith('.js') || file.endsWith('.jsx') || file.endsWith('.css')) {
            processFile(fullPath);
        }
    });
}

const srcDir = path.join(__dirname, 'src');
console.log('Starting migration...');
walk(srcDir);
console.log('Migration complete.');
